packages/algob/src/internal/deployer.ts

```diff
@@ -142,7 +142,25 @@ class DeployerBasicMode {
   }
 
   /**
-   * Description: Opt-In to ASA for a single account. The opt-in transaction is
+   * Appends signature (using signer's sk) to multi-signed logic signature. If multisig is not found
+   * then new multisig is created
+   * eg. appending own signature to a signed lsig (received from multisignature account address network)
+   * @param lsig Logic Sig object
+   * @param signer: Signer Account which will sign the smart contract
+   * @param mparams: passed when signing a new multisig
+   * @returns multi signed logic signature (with appended signature from signer's sk)
+   */
+  signMultiSig (lsig: LogicSig, signer: rtypes.Account, mparams?: MultisigMetadata): LogicSig {
+    if (lsig.msig === undefined) { // if multisig not found, create new msig
+      lsig.sign(signer.sk, mparams);
```

Comment:
what will happen if `mparams` are not passed ?

Reply:
and lsig is also not defined

Reply:
adding check for `mparams`. 
I would expect lsig to be passed (it is required), as what will user sign if lsig is not defined. Should we check for lsig as well?

Reply:
Sorry, I meant lsig.msig

Reply:
if `msig` is undefined then it means no account from multisig hash has signed this lsig yet. So "create" the msig in this case (hence we require the *mparams* in order). 

If lsig is already signed (by atleast one account of multisig) then msig should be present and we "append" our own signature to that msig (using `lsig.appendToMultisig(..)`).